### PR TITLE
Fix AirflowSkipException message raised by KubernetesPodOperator

### PR DESCRIPTION
**Pull request by dolfinus**
* `KubernetesPodOperator.skip_on_exit_code` is always a container (like `list[int]`), it cannot be `None`.
* If pod exited with matching exit code, `AirflowSkipException` should contain exact exit code instead of container value.
* Allow to pass `KubernetesPodOperator(skip_no_exit_code=0)` to skip successfully finished pods. Previously it was the same as `KubernetesPodOperator(skip_no_exit_code=[])`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/operators/pod.py
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/operators/pod.py]: This will not work if the user want to skip on exit code 0

**Review by hussein-awala — CHANGES_REQUESTED** [review]

File: airflow/providers/cncf/kubernetes/operators/pod.py
Comment (dolfinus) [reply airflow/providers/cncf/kubernetes/operators/pod.py]: Check again

File: airflow/providers/cncf/kubernetes/operators/pod.py
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/operators/pod.py:421]: 👍

**Review by hussein-awala — APPROVED** [review]
It looks good

File: airflow/providers/cncf/kubernetes/operators/pod.py
Comment (dolfinus) [reply airflow/providers/cncf/kubernetes/operators/pod.py]: Well, I have to change few more things to make this work

**Review by hussein-awala — APPROVED** [review]
I checked the recent changes, and they look good. Waiting for a second approval before merging.

**Comment by romsharon98** [comment]
Looks good!

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**
